File: ncarg2d/src/libncarg/stitle/ftitle.f
C
C $Id: ftitle.f,v 1.6 2008-07-27 00:17:26 haley Exp $
C                                                                      
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE FTITLE (MTST)
C
C The routine FTITLE reads from a "card input unit" groups of cards,
C each of which describes a single title frame that is to be displayed
C for a specified number of seconds using characters of a specified
C size.  FTITLE does the call to STITLE which is required to create the
C desired frames.
C
C FTITLE limits the number of lines on a given frame to "MCDS" (by
C default, 120).  If more lines than this are desired, reset the value
C in the following PARAMETER statement:
C
        PARAMETER (MCDS=120)
        CHARACTER*100 CRDS(MCDS)
C
C The common block SLCOMN holds all of the internal parameters of
C the package STITLE except for color-table parameters.
C
        COMMON /SLCOMN/ GPSZ,IBGC,IBGF,ICOP,IDOT,IFGC,IFGF,IJMP,IMAP,
     +                  INCU,IWLU,IWRK,IWWI,IXND,IXST,OORV,PCSZ,RNFS,
     +                  RVPB,RVPL,RVPR,RVPT,TFIN,TFOU,TGP1,TGP2,TGP3
        SAVE   /SLCOMN/
C
C Check for an uncleared prior error.
C
        IF (ICFELL('FTITLE - UNCLEARED PRIOR ERROR',1).NE.0) RETURN
C
C Set TGPN, which is the size of the gap preceding the next title (TGP1
C preceding the first title), and TGPF, which is the size of the final
C gap (zero so that, if no titles are found, we get no gap at all).
C
        TGPN=TGP1
        TGPF=0.
C
C Read up the first card of the next batch (done if end of file or zero
C card count are seen, error if there are too many title lines).
C
  101   NCDS=0
C
        READ (INCU,'(I5,2F5.1)',END=103,ERR=901) NCDS,TIME,SIZE
C
        IF (NCDS.LE.0) GO TO 103
C
        IF (NCDS.GT.MCDS) GO TO 902
C
C Compute the vertical size of a line and of the gap between lines and
C the sum of those.
C
        ISIZ=INT(SIZE*PCSZ)
        IGAP=INT(SIZE*GPSZ)
C
        ISUM=ISIZ+IGAP
C
C Set the X/Y coordinates of the first line.
C
        IXPS=64+448*ICOP
        IYPS=512+((NCDS-1)*ISUM)/2
C
C Set up the CRDS array for input to STITLE.
C
        DO 102 I=1,NCDS
          WRITE (CRDS(I)(1:20),'(3I5,F5.1)') IXPS,IYPS,ICOP,SIZE
          READ (INCU,'(A80)',ERR=902,END=903) CRDS(I)(21:100)
          IYPS=IYPS-ISUM
  102   CONTINUE
C
C Output a gap before calling STITLE.
C
        IF (TGPN.GT.0.) THEN
          CALL SLOGAP (TGPN,MTST)
          IF (ICFELL('FTITLE',2).NE.0) RETURN
        END IF
C
C Call STITLE to actually produce the desired frames.
C
        CALL STITLE (CRDS,NCDS,512,512,0.,TIME,0.,MTST)
        IF (ICFELL('FTITLE',3).NE.0) RETURN
C
C Reset TGPN and TGPF to the appropriate values for the gaps preceding
C the next label and following the last label.
C
        TGPN=TGP2
        TGPF=TGP2+TGP3
C
C Go back to read the next batch of cards.
C
        GO TO 101
C
C End of input detected.  Output a final gap.
C
  103   IF (TGPF.GT.0.) THEN
          CALL SLOGAP (TGPF,MTST)
          IF (ICFELL('FTITLE',4).NE.0) RETURN
        END IF
C
C Normal return.
C
        RETURN
C
C Error exits.
C
  901   CALL SETER ('FTITLE - READ ERROR ON CARD INPUT UNIT',5,1)
        RETURN
C
  902   CALL SETER ('FTITLE - TOO MANY INPUT CARDS IN GROUP',6,1)
        RETURN
C
  903   CALL SETER ('FTITLE - PREMATURE EOF ON CARD INPUT UNIT',7,1)
        RETURN
C
      END

File: ncarg2d/man/scrolled_title/ftitle.m
.TH FTITLE 3NCARG "July 1995" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
FTITLE - 
Used for unscrolled movie or video titles. It
reads, from standard input, the information necessary to
define the desired title frames, and then calls STITLE to
create those frames.
.SH SYNOPSIS
CALL FTITLE (MTST)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_ftitle (int mtst)
.SH DESCRIPTION 
.IP MTST 12
(an input expression of type INTEGER) is a switch
indicating whether this is a "real" run or a "practice" run:
.RS
.IP 0
means "real run".
.IP 1
means "practice run".
.RE
.IP ""
During real runs, each title frame is repeated as many
times as necessary to display it for a user-specified time
period (at \'NFS\' frames per second). Blank frames are placed
before the first title frame (\'TM1\' seconds worth of them), between
consecutive title frames (\'TM2\' seconds worth of them), and after
the last title frame (\'TM2\' + \'TM3\' seconds worth of them).
Blank frames are to allow for splicing.
.sp
If the user has turned on fade-in (by setting
the internal parameter \'FIN\' non-zero), and/or fade-out (by
setting the internal parameter \'FOU\' non-zero), the
required fade-in and fade-out frames will be generated, as
well.
.sp
During practice runs, each title frame will occur from one
to three times (once for fade-in, once for the stationary
holding time, and once for fade-out); each of these frames
will have a legend indicating how many seconds the frame
represents in a real run. Each sequence of blank frames
will be replaced by a single frame with a message
indicating how many seconds worth of blank frames it
represents.
.SH C-BINDING DESCRIPTION
The C-binding argument description is the same as the FORTRAN 
argument description.
.SH USAGE
FTITLE reads data from the unit specified by the internal parameter \'ICU\',
which has the default value 5, to specify standard input; this parameter
can be given a different value to specify that another unit should be read
instead.
.sp
The input data are read in groups.  Each group represents one title frame.
There can be any number of groups.  FTITLE keeps processing groups until
a group with NCDS = 0 is read or an end-of-file is encountered.  A group
consists of the following:
.IP \(bu 4
A header line from which variables NCDS, TIME, and SIZE are read, using the
FORTRAN format "(I5,2F5.1)".
.IP " " 4
NCDS is the number of text lines that follow.  If NCDS = 0, FTITLE quits
(returns to the calling routine without doing anything else).
.IP " " 4
TIME is the time, in seconds, that the title frame should be displayed (not
including fade-in and fade-out, if any).
.IP " " 4
SIZE is the desired character size, given as a multiplier of the default
height specified by the value of the internal parameter \'PSZ\', the
default value of which is 21 (out of 1023).  Values of SIZE from .75 to
2.5 are recommended.
.IP \(bu 4
Text lines, each containing one line of the title frame.  PLOTCHAR function
codes may be used as specified in the documentation for that package.
Characters should not appear beyond column 80.
.PP
The internal parameters 'ALN', 'BGB', 'BGC', 'BGF', 'BGG', 'BGR', 'FGB', 'FGC',
\&'FGF', 'FGG', 'FGR', 'FIN', 'FOU', 'GSZ', 'ICO', 'ICU', 'LOG', 'LX1', 'LX2',
\&'LY1', 'LY2', 'MAP', 'NFS', 'NXE', 'NXS', 'ORV', 'PSZ', 'SBK', 'SFG', 'TM1',
\&'TM2', 'TM3', 'VPB', 'VPL', 'VPR', 'VPT', and 'WID'
all affect the behavior of FTITLE in one way or another.  Some of these have
been mentioned above; all are described in the "man" page for
"scrolled_title_params".
.sp
Example: Suppose the input file contains the following three lines
.sp
.nf
    3   3.  1.5
A
Frame
of Titles
.fi
.sp
The resulting title frame has three lines of text.  It is displayed for
three seconds.  Characters have a size of 1.5 times the default character
size.
.sp
FTITLE allows a maximum of 80 characters per line of text including
Plotchar function codes.  No more than 120 lines of text can be
displayed on a single frame.  Titles are centered horizontally
unless you have changed the value of the internal parameter 'ICO'.
Vertical spacing is automatically determined using the current value of
the internal parameter 'GSZ'.
.sp
For more detailed control of titles, use the routine STITLE, which can be
used to generate either fixed or scrolled titles.
.SH EXAMPLES
Use the ncargex command to see the following relevant
example:
slex02.
.SH ACCESS
To use FTITLE or c_ftitle, load the NCAR Graphics libraries ncarg, ncarg_gks,
ncarg_c, and ncarg_c, preferably in that order.  
.SH MESSAGES
See the scrolled_title man page for a description of all Scrolled_title error
messages and/or informational messages.
.SH SEE ALSO
Online:
plotchar,
scrolled_title,
scrolled_title_params,
slgeti,
slgetr,
slogap,
slrset,
slseti,
slsetr,
stitle,
ncarg_cbind.
.sp
Hardcopy:
NCAR Graphics Fundamentals, UNIX Version;
User's Guide for NCAR GKS-0A Graphics
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
